Detectors/TRD/workflow/src/TRDTrapSimulatorSpec.cxx

```diff
@@ -0,0 +1,427 @@
+    //first check the directory listing.
+    LOG(warn) << " failed to get trapconfig from ccdb with name :  " << mTrapConfigName;
+    LOG(info) << "looking for TRAPconfig " << mTrapConfigName << __FILE__ << ":" << __LINE__;
+  }
+  LOG(info) << "looking for TRAPconfig " << mTrapConfigName << __FILE__ << ":" << __LINE__;
+  //  mTrapConfig = caltrap->Get(configName); //TODO it is not clear to me how this actually comes in.
+  //}
+  /// else {
+  //    if(mTrapConfig != nullptr){
+  //      delete mTrapConfig;
+  //    mTrapConfig = nullptr;
+  //  }
+  // mTrapConfig->LOADFROMDISKDEFAULT();
+  LOG(info) << "looking for TRAPconfig " << mTrapConfigName << __FILE__ << ":" << __LINE__;
+  LOG(warn) << "No TRAPconfig entry found for name of " << mTrapConfigName;
+  LOG(info) << "looking for TRAPconfig " << mTrapConfigName << __FILE__ << ":" << __LINE__;
+  //  }
+}
+
+void TRDDPLTrapSimulatorTask::init(o2::framework::InitContext& ic)
+{
+  //               LOG(debug) << "Input data file is : " << ic.options().get<std::string>("simdatasrc");
+  //               LOG(info) << "simSm is : " << ic.options().get<int>("simSM");
+  LOG(debug1) << "TRD Trap Simulator Device with pid of : " << ::getpid();
+  mfeeparam = FeeParam::instance();
+  mPrintTrackletOptions = ic.options().get<int>("printtracklets");
+  mDrawTrackletOptions = ic.options().get<int>("drawtracklets");
+  mShowTrackletStats = ic.options().get<int>("show-trd-trackletstats");
+  mTrapConfigName = ic.options().get<std::string>("trapconfig");
+  LOG(info) << "disable trap simualtor is : " << mDisableTrapSimulation;
+  LOG(info) << "Trap Simulator Device initialising with trap config of : " << mTrapConfigName;
+  //  if(mDisableTrapSimulation){
+  //  //now get a trapconfig to work with.
+  //    LOG(warn) << "You elected to not do a trap chip simulation and hence no trapconfig was saught";
+  //  }
+  //  else{
+  getTrapConfig();
+  LOG(info) << "Trap Simulator Device initialised ... ";
+  //  }
+}
+
+void TRDDPLTrapSimulatorTask::run(o2::framework::ProcessingContext& pc)
+{
+  LOG(info) << "TRD Trap Simulator Device running over incoming message ...";
+  if (mDisableTrapSimulation) {
+    LOG(warn) << " you elected to not do a trap chip simulation";
+    return;
+  }
+  // get the relevant inputs for the TrapSimulator
+  auto digits = pc.inputs().get<std::vector<o2::trd::Digit>>("digitinput");
+  //auto mMCLabels = pc.inputs().get<o2::dataformats::MCTruthContainer<o2::trd::MCLabel>*>("labelinput");
+  //auto mTriggerRecords = pc.inputs().get<std::vector<o2::trd::TriggerRecord>>("triggerrecords");
+
+  LOG(debug) << "Read in Digits with size of : " << digits.size();
+
+  //set up structures to hold the returning tracklets.
+  std::vector<Tracklet> mMCMTracklets; //vector to store the retrieved tracklets from an mcm
+  std::vector<Tracklet> mMCMTrackletsAccum;
+  mMCMTracklets.reserve(30);
+  mMCMTrackletsAccum.reserve(digits.size() / 3); //attempt to a. conserve mem, b. stop a vector resize
```

Comment:
These local variables (and also others) follow the naming convention of member variables, adds some confusion for people reading the code.  @bazinski maybe you can go through the code and change this.

Reply:
Will fix in the PR mentioned below.